node/common/client_config.ts

```diff
@@ -0,0 +1,339 @@
+export const CLIENT_PROPERTIES_FIELD = {
+  CERT_HOLDER_ID: 'scalar.dl.client.cert_holder_id',
+  CERT_VERSION: 'scalar.dl.client.cert_version',
+  CERT_PEM: 'scalar.dl.client.cert_pem',
+  PRIVATE_KEY_PEM: 'scalar.dl.client.private_key_pem',
+  PRIVATE_KEY_CRYPTOKEY: 'scalar.dl.client.private_key_cryptokey',
+  SERVER_HOST: 'scalar.dl.client.server.host',
+  SERVER_PORT: 'scalar.dl.client.server.port',
+  SERVER_PRIVILEGED_PORT: 'scalar.dl.client.server.privileged_port',
+  AUDITOR_ENABLED: 'scalar.dl.client.auditor.enabled',
+  AUDITOR_HOST: 'scalar.dl.client.auditor.host',
+  AUDITOR_PORT: 'scalar.dl.client.auditor.port',
+  AUDITOR_PRIVILEGED_PORT: 'scalar.dl.client.auditor.privileged_port',
+  AUDITOR_TLS_ENABLED: 'scalar.dl.client.auditor.tls.enabled',
+  AUDITOR_TLS_CA_ROOT_CERT_PEM: 'scalar.dl.client.auditor.tls.ca_root_cert_pem',
+  AUDITOR_LINEARIZABLE_VALIDATION_CONTRACT_ID:
+    'scalar.dl.client.auditor.linearizable_validation.contract_id',
+  TLS_CA_ROOT_CERT_PEM: 'scalar.dl.client.tls.ca_root_cert_pem',
+  TLS_ENABLED: 'scalar.dl.client.tls.enabled',
+  AUTHORIZATION_CREDENTIAL: 'scalar.dl.client.authorization.credential',
+  MODE: 'scalar.dl.client.mode',
+};
+
+export const CLIENT_MODE = {
+  CLIENT: 'CLIENT',
+  INTERMEDIARY: 'INTERMEDIARY',
+};
+
+export type Properties = {
+  [key: string]: string | number | boolean | Object;
+};
+
+function defaultProperties(): Properties {
+  const properties: Properties = {};
+
+  properties[CLIENT_PROPERTIES_FIELD.CERT_VERSION] = 1;
+  properties[CLIENT_PROPERTIES_FIELD.SERVER_HOST] = 'localhost';
+  properties[CLIENT_PROPERTIES_FIELD.SERVER_PORT] = 50051;
+  properties[CLIENT_PROPERTIES_FIELD.SERVER_PRIVILEGED_PORT] = 50052;
+  properties[CLIENT_PROPERTIES_FIELD.TLS_ENABLED] = false;
+  properties[CLIENT_PROPERTIES_FIELD.AUDITOR_ENABLED] = false;
+  properties[CLIENT_PROPERTIES_FIELD.AUDITOR_HOST] = 'localhost';
+  properties[CLIENT_PROPERTIES_FIELD.AUDITOR_TLS_ENABLED] = false;
+  properties[CLIENT_PROPERTIES_FIELD.AUDITOR_PORT] = 40051;
+  properties[CLIENT_PROPERTIES_FIELD.AUDITOR_PRIVILEGED_PORT] = 40052;
+  properties[
+    CLIENT_PROPERTIES_FIELD.AUDITOR_LINEARIZABLE_VALIDATION_CONTRACT_ID
+  ] = 'validate-ledger';
+  properties[CLIENT_PROPERTIES_FIELD.MODE] = CLIENT_MODE.CLIENT;
+
+  return properties;
+}
+
+function check(p: Properties) {
```

Comment:
This function makes sure that all required properties are set with the correct types.